src/components/contexts/confirmation-handler.tsx

```diff
@@ -0,0 +1,57 @@
+import { Divider, Typography } from "@mui/material";
+import { ReactNode } from "react";
+import GenericDialog from "../generics/generic-dialog";
+import strings from "../../localization/strings";
+
+/**
+ * Component properties
+ */
+interface Props {
+  children: ReactNode;
+  confirmation: string;
+  setConfirmation: (confirmation: string) => void;
+  deleteVacationsData: () => void;
+}
+
+/**
+ * Confirmation handler component
+ *
+ * @param props component properties
+ */
+const ConfirmationHandler = ({
+  children,
+  confirmation,
+  setConfirmation,
+  deleteVacationsData
+}: Props) => {
+  /**
+   * Component render
+   */
+  return (
+    <>
+      {children}
```

Comment:
Passing children so that you can wrap child components inside the confirmation handler component, however, this functionality is not needed so I deleted the children prop all together.